README.md

```diff
@@ -13,128 +13,339 @@
+
++ The _remove additional files at destination_ functionality provided by a `scm` (Kudu) deployment isn't supported by the deployment method used in this action. When you use [zip deploy](https://learn.microsoft.com/azure/azure-functions/functions-deployment-technologies#zip-deploy) or [one deploy for Flex Consumption plans](https://learn.microsoft.com/azure/azure-functions/functions-deployment-technologies#one-deploy), all files from a previous deployment are removed or are updated with the latest files from the deployment package. Any other files and directories outside of the deployment, such as added using FTP or created by your app during runtime, are preserved. If you're using `scm` with a package deployment, you must handle the removal of previously deployed files outside of the Functions action.
 
 ## Supported Languages and versions
-- [Language versions supported in each runtime version](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#languages-by-runtime-version)
-- [Languages supported in each OS](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#language-support-details)
 
-## Using Publish Profile as Deployment Credential (recommended)
-Using publish profile as deployment credential is recommended if you are not the owner of your Azure subscription. Follow these steps to configure your workflow to use the publish profile from your function app.
++ [Language versions supported in each runtime version](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#languages-by-runtime-version)
++ [Languages supported in each OS](https://docs.microsoft.com/en-us/azure/azure-functions/supported-languages#language-support-details)
+
+## Authentication methods
+
+You must choose how the action authenticates with Azure when deploying code from GitHub to your function app. There are three supported authentication methods:
+
+| Method| Description |
+| ---- | ---- |
+| [OpenID Connect (OIDC)](https://docs.github.com/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers) | This is the recommended and most secure authentication method from GitHub. Supports role-based access control (RBAC) for accessing Azure resources. Leverages user-assigned managed identities, which means that secrets are managed by Azure. |
+| Azure service principal | GitHub uses service principal credentials to access Azure resources using RBAC. You must manage the service principal secrets in GitHub. |  
+| Publish profile | Credentials (username and password) are stored in GitHub and used to access the `scm` endpoint using HTTP basic authentication during deployment. Might require you to enable the `scm` endpoint in your app. |
```

Comment:
We should link to the subsections in this section. We'll have to figure out a different way to surface the link to the GH docs - that is important to include. Maybe we just rely on the link that's already at the top of that subsection?

Reply:
Nice! I'll link to the subsections and figure out the best way to preserve the link to the GH docs.